apps/extension/src/provider/Signer.ts

```diff
@@ -58,6 +59,14 @@ export class Signer implements ISigner {
       };
     }
   }
+
+  public async sign(
+    signer: string,
+    data: string
+  ): Promise<SignatureResponse | undefined> {
+    return await this._namada.sign({ signer, data });
```

Comment:
This is just a question, but I get a bit confused about the difference between InjectedNamada and Signer. Is it correct that to call `sign`, it goes:

1. Call `getSigner` on InjectedNamada
2. Call `sign` on the returned Signer
3. ...which then calls `sign` on InjectedNamada
4. ...which calls the proxy to contact the extension?

Reply:
I think this is something that should be refactored in the near future, such that either all signing functions are only present in the `Signer` class, or they are moved to `Namada` and `Signer` is removed. That would apply to other functionality that is there. Originally we wanted to match the interface of Keplr's injected instance, but that's probably not as useful as we once thought.

Reply:
I opened this to track this issue: https://github.com/anoma/namada-interface/issues/583 - this is something that can be addressed before the next release